public/components/datasources/components/new/auth_details.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiFormRow, EuiFieldText, EuiFieldPassword } from '@elastic/eui';
+import { useState } from 'react';
+import React from 'react';
+import { AuthMethod } from '../../../../../common/constants/data_connections';
+
+interface AuthDetailProps {
+  currentAuthMethod: AuthMethod;
+  currentPassword: string;
+  currentUsername: string;
+  currentAccessKey?: string;
+  currentSecretKey?: string;
+  currentRegion?: string;
+  setRegionForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setAccessKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setSecretKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setPasswordForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setUsernameForRequest: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const AuthDetails = (props: AuthDetailProps) => {
+  const {
+    currentUsername,
+    currentPassword,
+    currentAccessKey,
+    currentSecretKey,
+    currentRegion,
+    currentAuthMethod,
+    setAccessKeyForRequest,
+    setPasswordForRequest,
+    setRegionForRequest,
+    setSecretKeyForRequest,
+    setUsernameForRequest,
+  } = props;
```

Comment:
At this point I would probably just use the object directly to make it more explicit, call it `authProps` and use e.g. `authProps.currentUsername` or `authProps.setAccessKeyForRequest`. The extra typing is worth the clarity.